src/Features/CSharp/Portable/AddBraces/CSharpAddBracesDiagnosticAnalyzer.cs

```diff
@@ -86,5 +108,172 @@ public void AnalyzeNode(SyntaxNodeAnalysisContext context)
                 properties: null,
                 SyntaxFacts.GetText(firstToken.Kind())));
         }
+
+        /// <summary>
+        /// <para>In general, statements are considered multiline if any of the following span more than one line:</para>
+        /// <list type="bullet">
+        /// <item><description>The part of the statement preceding the embedded statement</description></item>
+        /// <item><description>The embedded statement itself</description></item>
+        /// <item><description>The part of the statement following the embedded statement</description></item>
```

Comment:
Small suggestion (your call on if you want it).  Your doc comments are great for being concise and precise.  However, i think tehy could also benefit occasionally from a little more info in the form of an example.  For example (if i had not read the PR) i might go "Wait... what's 'The part of the statement following the embedded statement'?".  Having something simple like "For example, the `while (...)` clause in a `do ... while (...);` statement", goes a long way to grounding the statements.

It's like teh C# language spec in that way.  There's just the spec legalize, which should be simple and precise.  But there are also guiding examples to help ground the statements intuitively.  

--

Note: this example would be particularly relevant, because the most common example someone would think of would be an 'else' clause, btu then you specifically call that out as *not* counting in teh paragraph below.

Reply:
Even if you don't do this, the comment is super helpful.  Thanks for adding it!

Reply:
➡️ Now updated